WordPress/Classes/ViewRelated/Post/PostActionSheet.swift

```diff
@@ -60,8 +60,14 @@ class PostActionSheet {
                     actionSheetController.addDefaultActionWithTitle(Titles.cancelAutoUpload) { [weak self] _ in
                         self?.interactivePostViewDelegate?.cancelAutoUpload(post)
                     }
-                default:
-                    CrashLogging.logMessage("Cannot handle unexpected button for post action sheet: \(button). This is a configuration error.", level: .error)
+                case .retry:
+                    actionSheetController.addDefaultActionWithTitle(Titles.retry) { [weak self] _ in
+                        self?.interactivePostViewDelegate?.retry(post)
+                    }
+                case .edit:
+                    break // Not available in Action Sheet
```

Comment:
This feels like we're failing to decide if we support an "edit" action or not.

To put it in a different light, I think that here we are either:

- Silently ignoring an error for an invalid scenario, or;
- Missing proper support for a valid scenario.

My improvement suggestion here would be to either:

1. Implement the logic for the edit action and let the UI decide whether it's available or not, or;
2. Recording an error here through `CrashLogging.logError()` or `CrashLogging.logMessage()`.  The good thing about these methods is that they will log an error without having any visible effect for the user, while at the same time letting us know that a case we thought was invalid is not.

I think that option 1 may be best for the `.edit` case since the `InteractivePostViewDelegate` gives us a handy implementation for this case.

Option 2 could be better for the `.more` case since I'm not sure we have a handy implementation of it.

My 2 cents.

Reply:
Done in 2766441263b7490746e0592a32bf5963b11589e3